tests/python_client/testcases/test_query.py

```diff
@@ -654,6 +640,268 @@ def test_query_expr_json_contains_combined_with_normal(self, enable_dynamic_fiel
+            res = collection_w.query(expression)[0]
+            assert len(res) == 1
+
+            # test for string
+            ids = [str(_id), str(_id + 1), str(_id + 2)]
+            expression = f"{expr}(json_field['listStr'], {ids})"
+            res = collection_w.query(expression)[0]
+            assert len(res) == limit - len(ids) + 1
+
+            # test for float
+            ids = [_id * 1.0]
+            expression = f"{expr}(json_field['listFlt'], {ids})"
+            res = collection_w.query(expression)[0]
+            assert len(res) == limit
+
+            # test for bool
+            ids = [True]
+            expression = f"{expr}(json_field['listBool'], {ids})"
+            res = collection_w.query(expression)[0]
+            assert len(res) == ct.default_nb // 2
+
+            # test for list
+            ids = [[_id, str(_id + 1)]]
+            expression = f"{expr}(json_field['listList'], {ids})"
+            res = collection_w.query(expression)[0]
+            assert len(res) == 1
+
+            # test for mixed data
+            ids = [_id * 1.1, bool(_id % 2)]
+            expression = f"{expr}(json_field['listMix'], {ids})"
+            res = collection_w.query(expression)[0]
+            assert len(res) == 1
+
+    @pytest.mark.tags(CaseLabel.L2)
+    def test_query_expr_list_all_datatype_json_contains_all(self):
+        """
+        target: test query with expression using json_contains_all
+        method: query with expression using json_contains_all
+        expected: succeed
+        """
+        # 1. initialize with data
+        collection_w = self.init_collection_general(prefix, enable_dynamic_field=True)[0]
+
+        # 2. insert data
+        array = cf.gen_default_rows_data(with_json=False)
+        limit = 10
+        for i in range(ct.default_nb):
+            array[i]["listInt"] = [m for m in range(i, i + limit)]  # test for int
+            array[i]["listStr"] = [str(m) for m in range(i, i + limit)]  # test for string
+            array[i]["listFlt"] = [m * 1.0 for m in range(i, i + limit)]  # test for float
+            array[i]["listBool"] = [bool(i % 2)]  # test for bool
+            array[i]["listList"] = [[i, str(i + 1)], [i * 1.0, i + 1]]  # test for list
+            array[i]["listMix"] = [i, i * 1.1, str(i), bool(i % 2), [i, str(i)]]  # test for mixed data
+
+        collection_w.insert(array)
+
+        # 3. query
+        collection_w.load()
+        expr_prefix = ["json_contains_all", "JSON_CONTAINS_ALL"]
+        for expr in expr_prefix:
+            # test for int
+            _id = random.randint(0, ct.default_nb)
+            ids = [i for i in range(_id, _id + limit)]
+            expression = f"{expr}(listInt, {ids})"
+            res = collection_w.query(expression)[0]
+            assert len(res) == 1
+
+            # test for string
+            ids = [str(_id), str(_id + 1), str(_id + 2)]
+            expression = f"{expr}(listStr, {ids})"
```

Comment:
Should it be a list for the second parameter for json_contains_all or any?

Add more negative cases, and check the error message

Reply:
The invalid type and value check has been added in case test_query_expr_json_contains_invalid_type.